### ISSUE TITLE
Featured picture

### ISSUE THREAD

**Issue by Cryptonativa** [free2z/zuu#51]
Visually, it would be helpful when there is no featured photograph, a photograph from the page is automatically selected for the article preview.
![Screenshot 2023-04-11 at 8 31 37 PM](https://user-images.githubusercontent.com/128416795/231318407-24f5a07d-431f-4810-aba0-6e9b857d3799.png)

**Comment by skyl** [comment]
That particular page has no image associated with it but I get your meaning

**Comment by skyl** [comment]
Claimed for implementation on branch `fix/51-zpage-body-image`. I’ll add safe first-body-image selection for classic zpage cards ahead of the existing Tuzi fallback, with focused regression coverage.

**Comment by skyl** [comment]
Public-only implementation audit completed after the corrected PR #507 review. The current client PR is being closed because it cannot truthfully fix this from the list payload it receives.

### Reproduced production contract gap

The classic list path is `GET /api/zpage/` → `Find.tsx` → `PageListRow2`. The public `zPageListSerializer.get_content()` returns `description` when present; otherwise it renders the truncated body and strips HTML before returning the first 365 characters. Markdown image syntax therefore does not reach the card.

This was confirmed against the public API using zpage `07f4d5d1-0a09-4b55-b59c-4ebe277ca66c`:

- no featured image;
- detail payload contains a 6,553-character body with multiple Markdown images;
- list payload has `content === description` and no image syntax;
- the PR #507 helper returns the Tuzi fallback for the list object, although it finds `/uploadz/public/Zerodartz/zodl-yes1.webp` from the detail body.

### Why the public repo cannot safely complete the backend half today

The public tree contains `py/dj/apps/g12f/serializers/zpage.py`, but it does **not** contain:

- the `zPage` or `GenericFile` model definitions;
- zpage migrations;
- the zpage list/detail viewset and queryset that owns the `truncated_content` annotation/deferred fields;
- backend serializer/endpoint tests;
- an owned image-proxy contract.

Adding a serializer method that reads the full body would be untestable here and could introduce one deferred-field query per result. Returning arbitrary Markdown URLs would also promote creator-controlled remote beacons—including protocol-relative, loopback, link-local, and credential-bearing URLs—to every listing scroller.

I did not inspect private `tuzi` or any private data.

### Exact backend unblock

A complete fix needs an owned backend transaction with these contracts:

1. Derive and persist (or safely annotate without N+1 reads) the first eligible body preview whenever content is written/published.
2. Accept only verified public same-origin uploads such as an owned `/uploadz/public/.../` asset, or route through an explicit privacy-preserving image proxy. Do not expose arbitrary remote/loopback/link-local URLs in list previews.
3. Recompute/clear the preview when content changes or an upload stops being public, and backfill existing published pages.
4. Add the derived preview field to both real list and detail payloads while preserving `featured image > safe body preview > Tuzi fallback`.
5. Add model/queryset/serializer endpoint tests using realistic cases where list `content` is a description or stripped truncation but the full body contains an image; include unsafe-host rejection, stale preview clearing, and query-count coverage.
6. Then consume the field in both `PageListRow2` and `PageHeader`, using a real `<img>`/load-error path or layered background so failed loads recover to the dimmed Tuzi fallback.

Once that backend ownership and test surface is available publicly (or the backend contract lands in its owning repository), this issue becomes implementable end to end. Until then it is accurately `blocked` + `needs-backend`, not `in-progress`.
